### PR TITLE
Add BAP v2 (spatial reasoning + text LLMs + low-data regime + minecraft + synthetic data + new eval)

### PR DESCRIPTION
**Pull request by prashant-jayan21**
**BAP v2: An Enhanced Task Framework for Instruction Following in Minecraft Dialogues** (under review as a long journal paper; from UIUC)

[Linkedin post](https://www.linkedin.com/posts/prashantjayannavar_embodiedai-minecraft-syntheticdata-activity-7385764298037432320-9tP8?utm_source=share&utm_medium=member_desktop&rcm=ACoAAAvDCnkBlC194QHIlR9fZ4J9lrc-_NVAA8I) | [Thread on X](https://x.com/p_jayannavar/status/1974977705335836810?s=20) | [📝 Blog](https://www.alphaxiv.org/overview/2501.10836v3) (thanks alphaXiv) | [🤗 HF Paper](https://huggingface.co/papers/2501.10836)

Excited to see where the community takes this next. Feel free to reach out!

Developing interactive agents that can understand language, perceive their surroundings, and act within the physical world is a long-standing goal of AI research. The Minecraft Collaborative Building Task (MCBT) (Narayan-Chen, Jayannavar, and Hockenmaier 2019), a two-player game in which an Architect (A) instructs a Builder (B) to construct a target structure in a simulated 3D Blocks World environment, offers a rich platform to work towards this goal. In this work, we focus on the Builder Action Prediction (BAP) subtask: predicting B's actions in a multimodal game context (Jayannavar, Narayan-Chen, and Hockenmaier 2020) - a challenging testbed for grounded instruction following, with limited training data. We holistically re-examine this task and introduce BAP v2 to address key challenges in evaluation, training data, and modeling. Specifically, we define an enhanced evaluation benchmark, featuring a cleaner test set and fairer, more insightful metrics that also reveal spatial reasoning as the primary performance bottleneck. To address data scarcity and to teach models basic spatial skills, we generate different types of synthetic MCBT data. We observe that current, LLM-based SOTA models trained on the human BAP dialogues fail on these simpler, synthetic BAP ones, but show that training models on this synthetic data improves their performance across the board. We also introduce a new SOTA model, Llama-CRAFTS, which leverages richer input representations, and achieves an F1 score of 53.0 on the BAP v2 task and strong performance on the synthetic data. While this result marks a notable 6 points improvement over previous work, it also underscores the task's remaining difficulty, establishing BAP v2 as a fertile ground for future research, and providing a useful measure of the spatial capabilities of current text-only LLMs in such embodied tasks.

## Summary by CodeRabbit

* **Documentation**
  * Added BAP v2: An Enhanced Task Framework for Instruction Following in Minecraft Dialogues to both Visual-based and Text-based methods references with associated links, images, and code resources.

<sub>✏️ Tip: You can customize this high-level summary in your review settings.</sub>

### CODE REVIEW

**Comment by prashant-jayan21** [comment]
Hello @yyyybq. Any update on this?
Update on my end: the paper recently got accepted to the Computational Linguistics journal as a long paper. :)

In case the paper doesn't seem to fit the repo, feel free to close the pull request.

**Comment by yyyybq** [comment]
Thanks for the heads-up and congrats on the ACL journal acceptance—great news!
After an internal discussion we feel this work falls outside the repo’s current “spatial-intelligence” scope, so we’ll close the PR for now. Thanks again for understanding and best of luck with the journal publication!
